django/curator/management/commands/curator_spam_detection.py

```diff
@@ -0,0 +1,135 @@
+        )
+        parser.add_argument(
+            "--get_model_metrics",
+            "-g",
+            action="store_true",
+            default=False,
+            help="gets model accuracy, precision, recall and f1 scores",
+        )
+        parser.add_argument(
+            "--load_labels",
+            "-l",
+            action="store_true",
+            default=False,
+            help="save initial dataset to the DB. ",
+        )
+        parser.add_argument("--train_user", "-tu", action="store_true", default=False)
+        parser.add_argument(
+            "--p_train_user",
+            "-ptu",
+            action="store_true",
+            default=False,
+            help="perform partial train on data with no ML model recomendation",
+        )
+        parser.add_argument("--predict_user", "-pu", action="store_true", default=False)
+        parser.add_argument("--train_text", "-tb", action="store_true", default=False)
+        parser.add_argument("--predict_text", "-pb", action="store_true", default=False)
+
+    def handle_exe(self):
+        result = self.detection.execute()
+        print("Spam Users :\n", result.spam_users)
+        print(
+            "UserMetadataSpamClassifier Metircs :\n",
+            result.user_metadata_spam_classifier,
+        )
+        print("TextSpamClassifier  Metircs:\n", result.text_spam_classifier)
+
+    def handle_refine(self):
+        metrics = self.detection.refine()
+        print(
+            "UserMetadataSpamClassifier Metircs :\n",
+            metrics.user_metadata_spam_classifier,
+        )
+        print("TextSpamClassifier  Metircs:\n", metrics.text_spam_classifier)
+
+    def handle_get_model_metrics(self):
+        metrics = self.detection.get_model_metrics()
+        print(
+            "UserMetadataSpamClassifier Metircs :\n",
+            metrics.user_metadata_spam_classifier,
+        )
+        print("TextSpamClassifier  Metircs:\n", metrics.text_spam_classifier)
+
+    def handle_load_labels(self, load_directory):
+        self.processor.load_labels_from_csv(load_directory)
+
+    def handle_train_user(self):
+        self.user_meta_classifier.fit()
+
+    def handle_predict_user(self):
+        self.user_meta_classifier.predict()
+
+    def handle_p_train_user(self):
+        self.user_meta_classifier.partial_fit()
+
+    def handle_train_text(self):
+        self.text_classifier.fit()
+
+    def handle_predict_text(self):
+        self.text_classifier.predict()
+
+    def handle(self, *args, **options):
+        exe = options["exe"]
+        refine = options["refine"]
+        model_metrics = options["get_model_metrics"]
+        load = options["load_labels"]
+        train_user = options["train_user"]
+        predict_user = options["predict_user"]
+        p_train_user = options["p_train_user"]
+        train_text = options["train_text"]
+        predict_text = options["predict_text"]
+
+        load_directory = pathlib.Path(DATASET_FILE_PATH)
+
```

Comment:
could potentially shorten this to use dynamic method calls e.g. `getattr(self, f"handle_{action})()"`